src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator.py

```diff
@@ -0,0 +1,436 @@
+                        f"problems. You should use a flavor of "
+                        f"{stack_comp.TYPE.value} other than "
+                        f"'{stack_comp.FLAVOR}'.\n"
+                        + silence_local_validations_msg
+                    )
+
+                # if the orchestrator is remote, the container registry must
+                # also be remote.
+                if container_registry.is_local:
+                    return False, (
+                        f"The Kubernetes orchestrator is configured to run "
+                        f"pipelines in a remote Kubernetes cluster designated "
+                        f"by the '{self.kubernetes_context}' configuration "
+                        f"context, but the '{container_registry.name}' "
+                        f"container registry URI '{container_registry.uri}' "
+                        f"points to a local container registry. Please ensure "
+                        f"that you always use non-local stack components with "
+                        f"a remote Kubernetes orchestrator, otherwise you will "
+                        f"run into problems. You should use a flavor of "
+                        f"container registry other than "
+                        f"'{container_registry.FLAVOR}'.\n"
+                        + silence_local_validations_msg
+                    )
+
+            if not self.skip_local_validations and self.is_local:
+
+                # if the orchestrator is local, the container registry must
+                # also be local.
+                if not container_registry.is_local:
+                    return False, (
+                        f"The Kubernetes orchestrator is configured to run "
+                        f"pipelines in a local k3d Kubernetes cluster "
+                        f"designated by the '{self.kubernetes_context}' "
+                        f"configuration context, but the container registry "
+                        f"URI '{container_registry.uri}' doesn't match the "
+                        f"expected format 'localhost:$PORT'. "
+                        f"The local Kubernetes orchestrator only works with a "
+                        f"local container registry because it cannot "
+                        f"currently authenticate to external container "
+                        f"registries. You should use a flavor of container "
+                        f"registry other than '{container_registry.FLAVOR}'.\n"
+                        + silence_local_validations_msg
+                    )
+
+            return True, ""
+
+        return StackValidator(
+            required_components={StackComponentType.CONTAINER_REGISTRY},
+            custom_validation_function=_validate_local_requirements,
+        )
+
+    def get_docker_image_name(self, pipeline_name: str) -> str:
+        """Returns the full docker image name including registry and tag."""
+
+        base_image_name = f"zenml-kubernetes:{pipeline_name}"
```

Comment:
Cant we have a custom image instead?

Reply:
We can use custom base images by passing `custom_docker_base_image_name` to the `__init__()`. The code you're referring to actually constructs the final image name, not the base image name, so the variable name is misleading here.

This could ofc be easily improved by renaming the variable, but I won't do it yet, as there is a bigger underlying issue, which is that the whole function is duplicate code from the kubeflow integration, so the renaming would have to be done in several places to be consistent. IMO the proper way to fix this is to abstract away some of the duplicate code between the different remote orchestrators and then change it there.

Reply:
Renamed the variable.